### ISSUE TITLE
Adjust drools-ansible integration GHA

### ISSUE THREAD

**Issue by tkobayas** [apache/incubator-kie#6100]
For example
https://github.com/apache/incubator-kie-drools/actions/runs/11025843132/job/30621708911?pr=6099

```
2024-09-25T03:50:42.0123338Z [ERROR] [kiegroup/drools-ansible-rulebook-integration] Error cloning kiegroup/drools-ansible-rulebook-integration and switching to target branch 10.0.x
```

drools-ansible-rulebook-integration  doesn't have branch 10.0.x. We will need to adjust the configuration like `project-dependencies.yaml`

**Comment by tkobayas** [comment]
Hi @mariofusco , do we want to test drools `10.0.x` (so eventually 10.0.0) with any branch of `drools-ansible-rulebook-integration`?

Currently, using PR GHA, these combinations are tested.
- drools main & drools-ansible-rulebook-integration main
- drools 8.43.x & drools-ansible-rulebook-integration 1.0.x

If we don't plan to use  drools `10.0.x` at the moment, we may disable the GHA for 10.0.x. WDYT?

**Comment by mariofusco** [comment]
I believe that when drools 10.0.0 will be finally available (hopefully in no more than 2 or 3 weeks at this point) we may want to use it for the upcoming 1.0.x releases of drools-ansible-rulebook-integration. 

Also consider that if there will be the need of changing drools codebase (at the moment I'm trying to avoid this as much as possible as I did for the IsNotDefined fix, but such kind of workarounds could be not always possible of desirable) we will do so on the next 10.0.x releases while the 8.x stream has to be considered effectively closed.

For these reasons I think that we should keep (and fix in case it is broken) the GHA for 10.0.x and eventually get rid of the no longer useful one for 8.43.x when drools 10.0.0 will be out. Let me know if this is feasible or you see any issue.

**Comment by tkobayas** [comment]
This issue is more about `drools-ansible-rulebook-integration` rather than `incubator-kie-drools`. Moving the issue to
https://github.com/kiegroup/drools-ansible-rulebook-integration/issues/113
